spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+    chan_loc = we.recording.get_channel_locations()
+    unit_max_chan = get_template_extremum_channel(we, peak_sign=peak_sign, mode="extremum", outputs="index")
+    unit_max_chan = list(unit_max_chan.values())
+    unit_locations = chan_loc[unit_max_chan, :]
+    unit_distances = scipy.spatial.distance.cdist(unit_locations, unit_locations, metric='euclidean')
+    pair_mask = pair_mask & (unit_distances <=maximum_distance_um)
+    
+    print('Will check ', np.sum(pair_mask), 'pairs on ', pair_mask.size)
+    
+    # Step 4 : potential auto merge by correlogram
+    correlograms, bins = compute_correlograms(sorting, window_ms=window_ms, bin_ms=bin_ms, method='numba')
+    corr_diff = compute_correlogram_diff(sorting, correlograms, bins,
+                                    correlogram_low_pass=correlogram_low_pass,  adaptative_window_threshold=adaptative_window_threshold, 
+                                    pair_mask=pair_mask)
+    ind1, ind2 = np.nonzero(corr_diff  < corr_thresh)
+    potential_merges = list(zip(unit_ids[ind1], unit_ids[ind2]))
+    print(potential_merges)
+    
+    # step 5 : check if potential merge with CC also have template similarity
+    # TODO
+    
+    # step 6 : validate the potential merges with CC increase the contamination quality metrics
+    # TODO
+    
+    return potential_merges, corr_diff
+
+
+def compute_correlogram_diff(sorting, correlograms, bins,  correlogram_low_pass=800., adaptative_window_threshold=0.5,
+            pair_mask=None):
+    """
+    Original author: Aurelien Wyngaard ( lussac)
+    
+    Parameters
+    ----------
+    sorting: BaseSorting
+        The sorting object
+    correlograms: array 3d
+        The 3d array containing all cross and auto corrlogram
+    bins: array
+        Bins of the correlograms
+    correlogram_low_pass: float
+        low pass filter on correlogram for smoothing
+    adaptative_window_threshold
+        
+    pair_mask: None or boolean array
+        A bool matrix of size (num_units, num_units) to select
+        which pair to compute.
+    
+    Returns
+    -------
+    corr_diff
+    """
+    bin_ms = bins[1] - bins[0] 
+    
+    unit_ids = sorting.unit_ids
+    n = len(unit_ids)
+    
+    if pair_mask is None:
+        pair_mask = np.ones((n, n), dtype='bool')
+    
+    
+    # Index of the middle of the correlograms.
+    m = correlograms.shape[2] // 2
+    
+    num_spikes = sorting.get_total_num_spikes()
+    
+    # smooth correlogram by low pass filter
+    b, a = scipy.signal.butter(N=2, Wn = correlogram_low_pass / (1e3 / bin_ms /2), btype='low')
+    correlograms_smooth = scipy.signal.filtfilt(b, a, correlograms, axis=2)
+    
+    # find correlogram window for each units
+    win_sizes = np.zeros(n, dtype=int)
+    for unit_ind in range(n):
+        auto_corr = correlograms_smooth[unit_ind, unit_ind, :]
+        thresh = np.max(auto_corr) * adaptative_window_threshold
+        win_size = get_unit_adaptive_window(auto_corr, thresh)
```

Comment:
`get_unit_adaptive_window(auto_corr[m:], thresh)`
You only need half of the auto-corr (since it's symmetrical) and it's much easier to use (take the first peak = take the peak closest to the center).

Reply:
I prefer to make the cut inside the function to avoid mistake when calling.